src/tools/PICS-generator/PICSGenerator.py

```diff
@@ -0,0 +1,429 @@
+import argparse
+import json
+import os
+import pathlib
+import sys
+import xml.etree.ElementTree as ET
+
+import chip.clusters as Clusters
+from rich.console import Console
+
+# Add the path to python_testing folder, in order to be able to import from matter_testing_support
+sys.path.append(os.path.abspath(sys.path[0] + "/../../python_testing"))
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main  # noqa: E402
+
+console = None
+
+
+def GenerateDevicePicsXmlFiles(clusterName, clusterPicsCode, featurePicsList, attributePicsList, acceptedCommandPicsList, generatedCommandPicsList, outputPathStr):
+
+    xmlPath = xmlTemplatePathStr
```

Comment:
where does this come from?

Reply:
Of cause, my mistake. I will get this sorted.

Reply:
should it be a self.xmlTemplatePathStr

Reply:
xmlTemplatePathStr is the (checked) argument provided by --pics-template, not really sure why I do not just use xmlTemplatePathStr directly in this function. Let me give this a look and clean this up a bit 😄